src/append/file.rs

```diff
@@ -89,26 +90,56 @@ impl FileAppenderBuilder {
     /// will be resolved. Note that if the variable fails to resolve,
     /// $ENV{name_here} will NOT be replaced in the path.
     pub fn build<P: AsRef<Path>>(self, path: P) -> io::Result<FileAppender> {
-        let path_cow = path.as_ref().to_string_lossy();
-        let path: PathBuf = expand_env_vars(path_cow).as_ref().into();
-        if let Some(parent) = path.parent() {
+        let new_path = self.date_time_format(path);
+
+        if let Some(parent) = new_path.parent() {
             fs::create_dir_all(parent)?;
         }
         let file = OpenOptions::new()
             .write(true)
             .append(self.append)
             .truncate(!self.append)
             .create(true)
-            .open(&path)?;
+            .open(&new_path)?;
 
         Ok(FileAppender {
-            path,
+            path: new_path,
             file: Mutex::new(SimpleWriter(BufWriter::with_capacity(1024, file))),
             encoder: self
                 .encoder
                 .unwrap_or_else(|| Box::<PatternEncoder>::default()),
         })
     }
+
+    fn date_time_format<P: AsRef<Path>>(&self, path: P) -> PathBuf {
+        let path_cow = path.as_ref().to_string_lossy();
+        // Extract the environment path first
+        let env_path: PathBuf = expand_env_vars(path_cow).as_ref().into();
+
+        let date_time_path = env_path.to_str().unwrap();
+        // Locate the start and end of the placeholder
+        if let Some(start) = date_time_path.find("$TIME{") {
```

Comment:
Store your pattern as a constant so that you can use `.len()` to determine the length instead of hard coding it.

Reply:
Thanks for you review. I added const variables for TIME pattern:
```
const TIME_PREFIX: &str = "$TIME{";
const TIME_PREFIX_LEN: usize = TIME_PREFIX.len();
const TIME_SUFFIX: char = '}';
const TIME_SUFFIX_LEN: usize = 1;
```